From f2504eacb6d5e42b28df77041ff659fded3fd1a4 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 2 Nov 2014 12:43:31 +0100
Subject: [PATCH 3/5] Leave yacc defines to configure
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Resulted in:
* QA Notice: Package triggers severe warnings which indicate that it
*            may exhibit random runtime failures.
* lexer.c:1942:9: warning: implicit declaration of function ‘isatty’
* [-Wimplicit-function-declaration]
---
 configure.ac            | 2 ++
 lib/libsql/lexer.l      | 2 ++
 lib/libsql/sql_parser.h | 1 -
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index d80e7e5..30ea37a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -52,6 +52,8 @@ AC_PROG_MAKE_SET
 AC_FUNC_MKTIME
 AC_FUNC_STRTOD
 
+AC_CHECK_HEADERS_ONCE([unistd.h])
+
 dnl # *****************************************
 dnl # pkg-config check time
 dnl # *****************************************
diff --git a/lib/libsql/lexer.l b/lib/libsql/lexer.l
index fa8c5d1..44d65e8 100644
--- a/lib/libsql/lexer.l
+++ b/lib/libsql/lexer.l
@@ -1,4 +1,6 @@
 %{
+#include <config.h>
+
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
diff --git a/lib/libsql/sql_parser.h b/lib/libsql/sql_parser.h
index 2157e5c..0eba93f 100644
--- a/lib/libsql/sql_parser.h
+++ b/lib/libsql/sql_parser.h
@@ -2,7 +2,6 @@
 #define SQL_PARSER_H
 
 #include <glib.h>
-#define YY_NO_UNISTD_H
 
 
 typedef struct sql_statement        sql_statement;
-- 
2.1.2

